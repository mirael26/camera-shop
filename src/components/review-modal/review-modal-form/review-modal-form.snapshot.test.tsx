import { renderWithReduxAndRouter } from '../../../test/helpers/render-with-redux-and-router';
import ReviewModalForm from './review-modal-form';

test('ReviewModalForm component renders correctly', () => {
  const { asFragment } = renderWithReduxAndRouter(<ReviewModalForm onSuccess={ jest.fn() }/>);
  expect(asFragment()).toMatchSnapshot();
});
