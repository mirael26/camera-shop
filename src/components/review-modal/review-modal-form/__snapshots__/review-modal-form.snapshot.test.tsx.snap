// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewModalForm component renders correctly 1`] = `
<DocumentFragment>
  <p
    class="title title--h4"
  >
    Оставить отзыв
  </p>
  <div
    class="form-review"
  >
    <form
      data-testid="form"
      method="post"
    >
      <div
        class="form-review__rate"
      >
        <fieldset
          class="rate form-review__item"
          data-testid="rating-input-block"
        >
          <legend
            class="rate__caption"
          >
            Рейтинг
            <svg
              aria-hidden="true"
              height="9"
              width="9"
            >
              <use
                xlink:href="#icon-snowflake"
              />
            </svg>
          </legend>
          <div
            class="rate__bar"
          >
            <div
              class="rate__group"
            >
              <input
                class="visually-hidden"
                id="star-5"
                name="rate"
                type="radio"
                value="5"
              />
              <label
                class="rate__label"
                for="star-5"
                title="Отлично"
              />
              <input
                class="visually-hidden"
                id="star-4"
                name="rate"
                type="radio"
                value="4"
              />
              <label
                class="rate__label"
                for="star-4"
                title="Хорошо"
              />
              <input
                class="visually-hidden"
                id="star-3"
                name="rate"
                type="radio"
                value="3"
              />
              <label
                class="rate__label"
                for="star-3"
                title="Нормально"
              />
              <input
                class="visually-hidden"
                id="star-2"
                name="rate"
                type="radio"
                value="2"
              />
              <label
                class="rate__label"
                for="star-2"
                title="Плохо"
              />
              <input
                class="visually-hidden"
                id="star-1"
                name="rate"
                type="radio"
                value="1"
              />
              <label
                class="rate__label"
                for="star-1"
                title="Ужасно"
              />
            </div>
            <div
              class="rate__progress"
            >
              <span
                class="rate__stars"
                data-testid="review-form-rate"
              >
                0
              </span>
               
              <span>
                /
              </span>
               
              <span
                class="rate__all-stars"
              >
                5
              </span>
            </div>
          </div>
          <p
            class="rate__message"
          >
            Нужно оценить товар
          </p>
        </fieldset>
        <div
          class="custom-input form-review__item"
          data-testid="name-input-block"
        >
          <label>
            <span
              class="custom-input__label"
            >
              Ваше имя
              <svg
                aria-hidden="true"
                height="9"
                width="9"
              >
                <use
                  xlink:href="#icon-snowflake"
                />
              </svg>
            </span>
            <input
              name="user-name"
              placeholder="Введите ваше имя"
              type="text"
              value=""
            />
          </label>
          <p
            class="custom-input__error"
          >
            Нужно указать имя
          </p>
        </div>
        <div
          class="custom-input form-review__item"
          data-testid="advantage-input-block"
        >
          <label>
            <span
              class="custom-input__label"
            >
              Достоинства
              <svg
                aria-hidden="true"
                height="9"
                width="9"
              >
                <use
                  xlink:href="#icon-snowflake"
                />
              </svg>
            </span>
            <input
              name="user-plus"
              placeholder="Основные преимущества товара"
              type="text"
              value=""
            />
          </label>
          <p
            class="custom-input__error"
          >
            Нужно указать достоинства
          </p>
        </div>
        <div
          class="custom-input form-review__item"
          data-testid="disadvantage-input-block"
        >
          <label>
            <span
              class="custom-input__label"
            >
              Недостатки
              <svg
                aria-hidden="true"
                height="9"
                width="9"
              >
                <use
                  xlink:href="#icon-snowflake"
                />
              </svg>
            </span>
            <input
              name="user-minus"
              placeholder="Главные недостатки товара"
              type="text"
              value=""
            />
          </label>
          <p
            class="custom-input__error"
          >
            Нужно указать недостатки
          </p>
        </div>
        <div
          class="custom-textarea form-review__item"
          data-testid="review-input-block"
        >
          <label>
            <span
              class="custom-textarea__label"
            >
              Комментарий
              <svg
                aria-hidden="true"
                height="9"
                width="9"
              >
                <use
                  xlink:href="#icon-snowflake"
                />
              </svg>
            </span>
            <textarea
              minlength="5"
              name="user-comment"
              placeholder="Поделитесь своим опытом покупки"
            />
          </label>
          <div
            class="custom-textarea__error"
          >
            Нужно добавить комментарий
          </div>
        </div>
      </div>
      <button
        class="btn btn--purple form-review__btn"
        type="submit"
      >
        Отправить отзыв
      </button>
    </form>
  </div>
  <button
    aria-label="Закрыть попап"
    class="cross-btn"
    type="button"
  >
    <svg
      aria-hidden="true"
      height="10"
      width="10"
    >
      <use
        xlink:href="#icon-close"
      />
    </svg>
  </button>
</DocumentFragment>
`;
